File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let count=0;
    let candidate=0;
    
    for(let num of nums){
        if(count==0) candidate=num;
        count += (candidate == num)? 1: -1;
    }
    return candidate;
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cand = 0;
        int vote = 0;

        for(int num: nums){
            if(vote == 0) {
                cand = num;
                vote++;
            }
            else if(num == cand) vote++;
            else vote--;
        }

        // int count = 0;
        // for(int num: nums){
        //     if(num==cand){
        //         count++;
        //     }
        // }

        // if(count > nums.length/2) return cand;
        // return 

        return cand;
    }
}